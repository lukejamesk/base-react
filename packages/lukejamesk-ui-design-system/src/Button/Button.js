import React, { forwardRef } from 'react'
import PropTypes from 'prop-types'
import clsx from 'clsx'
import { ButtonBase, Typography } from '@lukejamesk/core'

const variantMapping = {
  default: '',
  primary: 'ljk-bg-primary ljk-hover:primary-dark ljk-text-white',
  secondary: 'ljk-text-primary',
  danger: 'ljk-bg-red-600 ljk-text-white',
}

export const buttonVariants = ['default', 'primary', 'secondary', 'danger']

const Button = forwardRef(({ variant, children, className, ...props }, ref) => {
  const v = variant || 'default'
  return (
    <ButtonBase
      className={clsx(variantMapping[v], 'ljk-p-xs ljk-rounded ljk-uppercase ljk-text-base')}
      ref={ref}
      {...props}
    >
      <Typography>{children}</Typography>
    </ButtonBase>
  )
})

Button.displayName = 'Button'

Button.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  variant: PropTypes.oneOf(buttonVariants),
}

export default Button
