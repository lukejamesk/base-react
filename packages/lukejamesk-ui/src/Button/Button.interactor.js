import ButtonBaseInteractor from '../ButtonBase/ButtonBase.interactor'

export default ButtonBaseInteractor
