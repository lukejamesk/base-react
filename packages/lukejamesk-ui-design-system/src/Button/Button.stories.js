import React from 'react'
import { text } from '@storybook/addon-knobs'
import Button from './Button'

export default {
  title: 'Design System/Button',
  component: Button,
}

export const Default = () => <Button>{text('text', 'My Button')}</Button>
export const Primary = () => <Button variant="primary">{text('text', 'My Button')}</Button>
export const Secondary = () => <Button variant="secondary">{text('text', 'My Button')}</Button>
export const Danger = () => <Button variant="danger">{text('text', 'My Button')}</Button>
