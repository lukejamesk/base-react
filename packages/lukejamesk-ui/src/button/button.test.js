import React from 'react'
import { mount } from 'enzyme'
import Button from './button'
import { defaultTheme, ThemeProvider } from '@lukejamesk-styles'

describe('<Button />', () => {
  it('should render', () => {
    const wrapper = mount(
      <ThemeProvider theme={defaultTheme}>
        <Button>
          My content for button
        </Button>
      </ThemeProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
