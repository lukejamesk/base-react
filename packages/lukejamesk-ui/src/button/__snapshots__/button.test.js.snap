// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render 1`] = `
<ThemeProvider
  theme={
    Object {
      "palette": Object {
        "primary": "#39c5ff",
        "secondary": "#5c946e",
        "text": "rgba(0, 0, 0, .9)",
      },
      "spacing": Object {
        "l": 20,
        "m": 16,
        "s": 12,
        "xl": 24,
        "xs": 8,
        "xxl": 28,
        "xxs": 4,
      },
      "typography": Object {
        "bold": 600,
        "button": Object {
          "fontFamily": "Verdana, Arial, Sans-serif",
          "fontSize": 14,
        },
        "fontFamily": "Helvetica, Arial, Sans-serif",
        "fontSize": 14,
        "heading1": Object {
          "fontFamily": "Verdana, Arial, Sans-serif",
          "fontSize": 25,
          "fontWeight": 600,
        },
        "heading2": Object {
          "fontFamily": "Verdana, Arial, Sans-serif",
          "fontSize": 21,
          "fontWeight": 600,
        },
        "heading3": Object {
          "fontFamily": "Verdana, Arial, Sans-serif",
          "fontSize": 18,
          "fontWeight": 600,
        },
      },
    }
  }
>
  <Button>
    <button
      className="button-0-2-1"
      type="button"
    >
      <Typography>
        <span
          className="all-0-2-4"
        >
          My content for button
        </span>
      </Typography>
    </button>
  </Button>
</ThemeProvider>
`;
